AK/Time.cpp

```diff
@@ -283,10 +283,13 @@ MonotonicTime MonotonicTime::now_coarse()
 UnixDateTime UnixDateTime::from_iso8601_week(u32 week_year, u32 week)
 {
     auto january_1_weekday = day_of_week(week_year, 1, 1);
-    i32 offset_to_monday = (january_1_weekday <= 3) ? -january_1_weekday : 7 - january_1_weekday;
-    i32 first_monday_of_year = 1 + offset_to_monday;
+    i32 first_monday_of_year = 1 - january_1_weekday;
     i32 day_of_year = (first_monday_of_year + (week - 1) * 7) + 1;
 
+    if (day_of_year < 0) {
+        return UnixDateTime::from_iso8601_week(week_year - 1, weeks_in_year(week_year - 1) + 1);
+    }
```

Comment:
In your commit message you wrote:

> in the HTML spec, the first week is the earliest week that overlaps with the year.

That's not what ISO-8601 weeks represent; that first week was defined as the one containing the first Thursday of January (or the first week with at least 4 days of January in it).

Reply:
We could do with some tests for this method in TestTime.cpp, especially with inputs around weeks 52, 53 and 1.